Title:
Identifying Best Application to attract More Users.
Introduction:
The Project is aimed at helping developers understand what type of free apps that are likely to attract more users in order to generate more reveneus.
The Project will basically be aimed at Applications that are free todownload on Google pla and App Store where the only source of revenue is the in-apps ads


In [1]:
from csv import reader
def opening_file(file, header=True):
    opened_file=open(file)
    read_file = reader(opened_file)
    apps_data= list(read_file)
    if header:
        return apps_data[1:], apps_data[0]
   
    return apps_data


In [2]:
apple_file=opening_file(file='AppleStore.csv')
android_file = opening_file(file = 'googleplaystore.csv')

In [3]:
def explore_data(dataset, start, end, rows_and_column=False):
    dataset_slice=dataset[start:end]
    for row in dataset_slice:
        print (row)
        print('\n')
    if rows_and_column:
        print('Number of rows:', len(dataset))
        print("\n")
        print('Number of Columns:', len(dataset[0]))
    

In [4]:
def printheader(data):
    return data[1]
def excludeheader(data):
    return data[0]

In [5]:
applestore_data = excludeheader(data=apple_file)
googlestore_data = excludeheader(data=android_file)

In [6]:
explore_data(dataset=applestore_data, start=0, end=1, rows_and_column=True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


Number of rows: 7197


Number of Columns: 16


In [7]:
headerplay=printheader(data=apple_file)
headerdroid=printheader(data=android_file)
print(headerplay)
print('\n')
print(headerdroid)


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [8]:
print(googlestore_data[10472])
del googlestore_data[10472]


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


To Analyze these Datas, we need to make sur that there are no duplicates. My next approach here is to remove the Applications that have duplicates as seen with instagram as an example below.

In [9]:
for app in googlestore_data:
    name = app[0]
    if name == 'Instagram':
        print (app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [10]:
unique_data = []
duplicates_data = []
for row in googlestore_data:
    name = row[0]
    if name in unique_data:
        duplicates_data.append(name)
    else:
        unique_data.append(name)
print('Number of Duplicate Datas:', len(duplicates_data)) 
print('\n')
print('Examples of Duplicate datas:', duplicates_data[:15])

Number of Duplicate Datas: 1181


Examples of Duplicate datas: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


As seen above, there are 1181 duplicates datas. To work on the datas, duplicate datas will be removed but not randomly. Instead, the criteria will be that, the application with the highest rating will be left to work on

In [11]:
print('Expected length:', len(googlestore_data)-1181)

Expected length: 9659


In [12]:
reviews_max = {}
for app in googlestore_data:
    name = app[0]
    n_reviews = float(app[3])
    if (name in reviews_max) and (reviews_max[name] < n_reviews):
        reviews_max[name]=n_reviews
    if name not in reviews_max:
        reviews_max[name]= n_reviews
print (len(reviews_max))

9659


In [13]:
googlestore_clean = []
added_app = []
for app in googlestore_data:
    name = app[0]
    n_reviews = float(app[3])
    if (reviews_max[name] == n_reviews) and (name not in added_app):
        googlestore_clean.append(app)
        added_app.append(name)



In [14]:
explore_data(googlestore_clean, 50, 55, True)

['Offroad Oil Tanker Driver Transport Truck 2019', 'AUTO_AND_VEHICLES', '4.3', '542', '33M', '100,000+', 'Free', '0', 'Everyone', 'Auto & Vehicles', 'July 31, 2018', '4.0', '4.1 and up']


['Tickets SDA 2018 and Exam from the State Traffic Safety Inspectorate with Drom.ru', 'AUTO_AND_VEHICLES', '4.9', '10479', '33M', '100,000+', 'Free', '0', 'Everyone', 'Auto & Vehicles', 'July 18, 2018', '1.7.1', '4.0 and up']


['Gas Prices (Germany only)', 'AUTO_AND_VEHICLES', '4.4', '805', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Auto & Vehicles', 'July 29, 2018', '2.5.1', '4.4 and up']


['Extreme Rally Championship', 'AUTO_AND_VEHICLES', '4.2', '129', '54M', '100,000+', 'Free', '0', 'Everyone', 'Auto & Vehicles', 'July 26, 2018', '3.0', '4.1 and up']


['Restart Navigator', 'AUTO_AND_VEHICLES', '4.0', '1403', '201k', '100,000+', 'Free', '0', 'Everyone', 'Auto & Vehicles', 'August 26, 2014', '1.0.1', '2.2 and up']


Number of rows: 9659


Number of Columns: 13


In [15]:
def string_iterate(data):
    non_ascii = 0
    for character in data:
        if ord(character) > 127:
            non_ascii +=1
    if non_ascii > 3:
            return False
    else:
            return True

In [16]:
google_english= []
google_non_english =[]
apple_english = []
apple_non_english = []

for app in googlestore_clean:
    name = app[0]
    if string_iterate(name):
        google_english.append(app)
    else:
        google_non_english.append(app)
for app in applestore_data:
    name = app[1]
    if string_iterate(name):
        apple_english.append(app)
    else:
        apple_non_english.append(app)
explore_data(google_english,0, 2, True)
explore_data(apple_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9614


Number of Columns: 13
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183


Number of Columns: 16


In [17]:
apple_free_data = []
google_free_data=[]
for app in apple_english:
    price = app[4]
    if price == '0.0':
        apple_free_data.append(app)
for app in google_english:
    price = app[7]
    if price == '0':
        google_free_data.append(app)
explore_data(apple_free_data, 0, 5, True)
explore_data(google_free_data, 0, 5 , True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 3222


Number of Columns: 16
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 

The Aim of this Project is to Build the kinds of apps that is likely more used by a lot of users in order to help increase revenue:

Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

In [18]:
print(headerplay)
print('\n')
print(headerdroid)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [19]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total+=1
        value = row[index]
        if value in table:
            table[value]+=1
        else :
            table[value] = 1
    table_percentages = {}
    for key in table:
        percentages = (table[key]/total)*100
        table_percentages[key] = percentages
    return table_percentages
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
apple_prime_genre = display_table(apple_free_data, 11)
print('\n')
google__category= display_table(google_free_data, 1)
print('\n')
google__genres = display_table(google_free_data, 9)
    

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
C

We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.



In [20]:
genres_ios = freq_table(apple_free_data, 11)
apple={}
for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in apple_free_data:
        genre_app = app[11]
        if genre_app == genre:
            n_rating = float(app[5])
            total += n_rating
            len_genre += 1
    avg_number = total/len_genre
    apple[genre]= avg_number
for data in apple:
    datas = apple[data]
    print(data, ';', datas)
    print('\n')
    

Social Networking ; 71548.34905660378


Photo & Video ; 28441.54375


Games ; 22788.6696905016


Music ; 57326.530303030304


Reference ; 74942.11111111111


Health & Fitness ; 23298.015384615384


Weather ; 52279.892857142855


Utilities ; 18684.456790123455


Travel ; 28243.8


Shopping ; 26919.690476190477


News ; 21248.023255813954


Navigation ; 86090.33333333333


Lifestyle ; 16485.764705882353


Entertainment ; 14029.830708661417


Food & Drink ; 33333.92307692308


Sports ; 23008.898550724636


Book ; 39758.5


Finance ; 31467.944444444445


Education ; 7003.983050847458


Productivity ; 21028.410714285714


Business ; 7491.117647058823


Catalogs ; 4004.0


Medical ; 612.0




In [21]:
google_category = freq_table(google_free_data, 1)
google={}
for category in google_category:
        total = 0
        len_category = 0
        for app in google_free_data:
            category_app= app[1]
            if category_app == category:
                n_installs = app[5]
                n_installs=n_installs.replace('+', '')
                n_installs=n_installs.replace(',', '')
                n_installs=n_installs = float(n_installs)
                total += n_installs
                len_category += 1
        avag_number = total/len_category
        google[category] = avag_number


In [22]:

for app in google:
    yes =  google[app]
    print (app , ';', yes)
    print('\n')
print (len(google))


ART_AND_DESIGN ; 1986335.0877192982


AUTO_AND_VEHICLES ; 647317.8170731707


BEAUTY ; 513151.88679245283


BOOKS_AND_REFERENCE ; 8767811.894736841


BUSINESS ; 1712290.1474201474


COMICS ; 817657.2727272727


COMMUNICATION ; 38456119.167247385


DATING ; 854028.8303030303


EDUCATION ; 1833495.145631068


ENTERTAINMENT ; 11640705.88235294


EVENTS ; 253542.22222222222


FINANCE ; 1387692.475609756


FOOD_AND_DRINK ; 1924897.7363636363


HEALTH_AND_FITNESS ; 4188821.9853479853


HOUSE_AND_HOME ; 1331540.5616438356


LIBRARIES_AND_DEMO ; 638503.734939759


LIFESTYLE ; 1437816.2687861272


GAME ; 15588015.603248259


FAMILY ; 3695641.8198090694


MEDICAL ; 120550.61980830671


SOCIAL ; 23253652.127118643


SHOPPING ; 7036877.311557789


PHOTOGRAPHY ; 17840110.40229885


SPORTS ; 3638640.1428571427


TRAVEL_AND_LOCAL ; 13984077.710144928


TOOLS ; 10801391.298666667


PERSONALIZATION ; 5201482.6122448975


PRODUCTIVITY ; 16787331.344927534


PARENTING ; 542603.6206896552


WEATHER ; 5074

As displayed above, Communication apps has the highest number of downloads.
From Apple store data set, Social Networking has the highest number of downloads. 
Therefore, Our recoomendation after anylysing this data is that developers should develop a social networking and communication application 